src/main/java/com/github/dockerjava/api/model/SwarmInfo.java

```diff
@@ -0,0 +1,141 @@
+package com.github.dockerjava.api.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+import java.util.*;
+
+/**
+ *
+ * @author Michael Freund (fragger@xfragger.de)
+ *
+ */
+@JsonInclude(Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class SwarmInfo extends Info {
+
+    @SuppressWarnings("unchecked")
+    private static List<String> castLine(Object line) {
+        return (List<String>)line;
+    }
+
+    @JsonIgnore
+    public Map<String, Node> getSwarmNodes() {
+        final Map<String, Node> nodes = new HashMap<String, Node>();
+
+        boolean nodesFound = false;
+        Node lastNode = null;
+
+        for(Object lineObj : getDriverStatuses()) {
+            List<String> lineArr = castLine(lineObj);
+
+            //these strings have control chars (backspace as first char for example), remove all of them
+            String key = lineArr.get(0).replaceAll("[\u0000-\u001f]", "");
+
+            if (key.equals("Nodes")) {
+                nodesFound = true;
+                continue;
+            }
+
+            if (nodesFound && !key.substring(0, 3).equals(" └ ")) {
+                Node node = new Node(key, lineArr.get(1));
+                nodes.put(key, node);
+                lastNode = node;
+                continue;
+            }
+
+            if (lastNode != null) {
+                switch (key) {
+                    case " └ Status":
```

Comment:
it IS terrible... but the api answers with an backspace before these  └  signs....